framework/src/utils/PetscDMMoose.C

```diff
@@ -140,7 +140,7 @@ DMMooseValidityCheck(DM dm)
                      "Got DM of type %s, not of type %s",
                      ((PetscObject)dm)->type_name,
                      DMMOOSE);
-  PetscFunctionReturn(0);
+  return PetscErrorCode(0);
```

Comment:
Should we just be doing `return PETSC_SUCCESS` for all these?  More self-documenting IMHO.

Reply:
That works for me because I do not care to be compliant with old PETScs. I was reticent to do this in libMesh because I figure you and @jwpeterson may want to be more strict about being compatible with old versions. `PETSC_SUCCESS` was first introduced in 2022 I believe

Reply:
2022?  Whoa; I didn't realize that.

Reply:
I'm fine either way here, then.